app/models/allocation.rb

```diff
@@ -156,4 +159,18 @@ def prison_fix(movement_type)
             :event,
             :event_trigger,
             :prison, presence: true
+
+private
+
+  def check_push
```

Comment:
This could do with a comment explaining why the push is done is 2 pieces

Reply:
We should be able to use Rails' Dirty Tracking in the `after_save` without needing to capture the dirty state in a `before_save`.

Rather than using `primary_pom_nomis_id_changed?`, we can use `saved_change_to_primary_pom_nomis_id?` to tell us whether the most recent save changed the `primary_pom_nomis_id`.

This would allow us to simplify the implementation by removing the `before_save` callback and `@do_push` instance variable, and instead perform the conditional check directly in the `after_save` callback / `push_pom_to_community_api` method.

See the docs for more info: https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-saved_change_to_attribute-3F

@Katy600 it would be good to give this a try, as it feels cleaner than using two callbacks. If not possible, then a comment to explain why it's done in 2 pieces is a good idea (as suggested by @starswan).